net: lib: download_client: Allow use of DTLS CID (#12412)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/download_client.rst

The application must provision the TLS credentials and pass the security tag to the library when using CoAPS and calling :c:func:`download_client_connect`.

> When modem firmware v1.3.5 or newer is used, you can enable the support for DTLS Connection Identifier feature in the download client library by enabling :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_CID` Kconfig option.

Reviewer: ```suggestion
When modem firmware v1.3.5 or newer is used, you can enable the support for DTLS Connection Identifier feature in the download client library by enabling the :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_CID` Kconfig option.
```

> When modem firmware v1.3.5 or newer is used, you can enable the support for DTLS Connection Identifier feature in the download client library by enabling the :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_CID` Kconfig option.

Reviewer 2: ```suggestion
When you have modem firmware v1.3.5 or newer, you can use the :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_CID` Kconfig option to enable the DTLS Connection Identifier feature in this library.
```
